template/web/wp-content/mu-plugins/studiometa-maintenance-mode/class-studiometamaintenancemode.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Display maintenance page when maintenace is enabled.
+ *
+ * @package    studiometa/create-wordpress-project
+ * @author     Studio Meta <agence@studiometa.fr>
+ * @copyright  2021 Studio Meta
+ * @license    https://opensource.org/licenses/MIT
+ * @since      1.0.0
+ * @version    1.0.0
+ */
+
+/**
+ * StudiometaMaintenanceMode class.
+ */
+class StudiometaMaintenanceMode {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( 'init', array( $this, 'redirect_to_maintenance' ) );
+	}
+
+	/**
+	 * Is maintenance?
+	 *
+	 * @return boolean Is maintenance?
+	 */
+	protected function is_maintenance_mode() {
+		if (
+			'true' !== getenv( 'MAINTENANCE_ENABLED' )
+			|| is_admin()
+			|| current_user_can( 'administrator' )
+			|| ! file_exists( __DIR__ . '/maintenance.php' )
+			|| 'wp-login.php' === $GLOBALS['pagenow']
+		) {
+			return false;
+		}
+
+		$allowed_ips = explode( ',', getenv( 'MAINTENANCE_IPS' ) );
+		$remote_ip   = isset( $_SERVER['REMOTE_ADDR'] ) ? filter_var( wp_unslash( $_SERVER['REMOTE_ADDR'] ), FILTER_VALIDATE_IP ) : false;
+
+		if ( in_array( $remote_ip, $allowed_ips, true ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Redirect to maintenance mode.
+	 *
+	 * @return void
+	 */
+	public function redirect_to_maintenance() {
+		if ( ! $this->is_maintenance_mode() ) {
+			return;
+		}
+
+		$protocol = isset( $_SERVER['SERVER_PROTOCOL'] ) && 'HTTP/1.1' === $_SERVER['SERVER_PROTOCOL'] ? 'HTTP/1.1' : 'HTTP/1.0';
+
+		header( $protocol . ' 503 Service Unavailable', true, 503 );
```

Comment:
@lusimeon @studiometa/wordpress To me, there is a problem with this implementation of the maintenance page. If, the site is under maintenance and Google Robots come to browse the site in the meantime, pages of the site will be referenced with a 503 and the wrong content. This should not be the case and I think from the SEO perspective, we should redirect the visitor to a new page with a 302, this behaviour won't impact the SEO of the page. This needs to be discussed with Geoffrey to be sure!